vignettes/04-synthesis-data.Rmd

```diff
@@ -1 +1,294 @@
+```
+
+## Create histogram of growth rate for all cultivars
+
+The last thing that we are going to do is assess the difference in this relationship among the cultivars. 
+We are going to use the inflection point from the logistic growth model, which indicates when canopy cover stops increasing as quickly with increasingly more warm days. 
+The resulting inflection points for each cultivar are plotted as a histogram. 
+
+```{r plot_inflections}
+ggplot(all_cultivars) +
+  geom_point(aes(x = gdd_cum, y = mean)) +
+  geom_line(aes(x = gdd_cum, y = mean_predict), color = "orange") +
+  geom_vline(aes(xintercept = inf_point)) +
+  facet_wrap(~cultivar, scales = "free_y") +
+  labs(x = "Cumulative growing degree days", y = "Canopy Height")
+ggplot(data.frame(inf_points = unique(all_cultivars$inf_point))) +
+  geom_histogram(aes(x = inf_points)) +
+  xlim(min(all_cultivars$gdd_cum), max(all_cultivars$gdd_cum)) +
+  labs(x = "Inflection points", y = "Number")
+```
+
+## Get image data
+
+In this examnple we will extract our plot data from a series of images taken in May of Season 6, measure its "greeness" annd plot that against the plant heights from above in this vignette.
+
+Below we retrieve all the available plots for a particular date, then find and convert the plot boundary JSON into tuples.
+We will use these tuples to extract the data for our plot.
+
+```{r get_plot_boundary}
+library(traits)
+library(stringr)
+
+# Function for breaking apart a corner into its Lat, Lon components
+getLatLon <- function(corner){
+  p <- strsplit(corner, ' ')
+  return (c(p[[1]][1], p[[1]][2]))
+}
+
+# Gets the bounding box of the array of points
+getBounds <-function(bounds){
+  minX <- NA
+  minY <- NA
+  maxX <- NA
+  maxY <- NA
+  for (c in unique(bounds)){
+    p = getLatLon(c)
+    if (is.na(minX) || (minX > p[2]))
+      minX = p[2]
+    if (is.na(minY) || (minY > p[1]))
+      minY = p[1]
+    if (is.na(maxX) || (maxX < p[2]))
+      maxX = p[2]
+    if (is.na(maxY) || (maxY < p[1]))
+      maxY = p[1]
+  }
+  
+  return (c(minX, minY, maxX, maxY))
+}
+
+# Setting up our options
+options(betydb_url = "https://terraref.ncsa.illinois.edu/bety/",
+        betydb_api_version = 'v1',
+        betydb_key = readLines('.betykey'))
+
+# Makiong the query for our site
+sites <- betydb_query(table     = "sites",  
+                      sitename  = "MAC Field Scanner Season 6 Range 19 Column 1")
+
+# Assigning the geometry of the site (GeoJSON format)
+site.geom <- sites$geometry
+
+# Stripping out the extra information to get to the points
+complete_str <- str_match_all(site.geom, '(\\(\\(\\((.*)\\)\\)\\))')[[1]][, 3]
+bounds <- strsplit(complete_str, ', ')[[1]]
+
+# Getting the bounding box of the polygon
+bounding_box = getBounds(bounds)
+```
+
+These are the names of the full field RGB data for the month of May.
+We will be extracting our plot data from these files.
+A compressed file containing these images can be found on [Box]().
```

Comment:
"Box" will be the link to the tar containing the .tif files?

Reply:
It's filled in now. A reference to google drive with ZIP file